static/js/stream_list.js

```diff
@@ -214,6 +217,12 @@ export function get_stream_li(stream_id) {
     return $li;
 }
 
+export function build_add_stream_link() {
+    const $subscribe_to_more_streams_link_element = $("#subscribe-to-more-streams");
+    $subscribe_to_more_streams_link_element.empty();
+    $subscribe_to_more_streams_link_element.append(render_subscribe_to_more_streams);
```

Comment:
You need to actually call the function here, not just pass a function to `.append()`. 

In general, please actually test your intermediate commits to make sure they aren't broken in this way -- this sort of carelessness makes life difficult for reviewers.